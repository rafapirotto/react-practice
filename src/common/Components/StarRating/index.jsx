import React from 'react';

import semiFilledStar from './assets/semi-filled-star.png';
import filledStar from './assets/filled-star.png';

const renderFilledStars = (numberOfStars) => {
  let filledStars = [];

  for (let index = 0; index < numberOfStars; index++) {
    filledStars.push(<img src={filledStar} alt="filled-star" className="px-1" />);
  }

  return filledStars;
};

const renderHalfFilledStars = (numberOfStars) => {
  let semiFilledStars = [];

  for (let index = 0; index < numberOfStars; index++) {
    semiFilledStars.push(<img src={semiFilledStar} alt="half-filled-star" className="px-1" />);
  }

  return semiFilledStars;
};

const StarRating = ({ filledStars, halfFilledStars }) => {
  return (
    <div className="ml-12 flex items-center">
      {renderFilledStars(filledStars)}
      {renderHalfFilledStars(halfFilledStars)}
    </div>
  );
};

export default StarRating;
