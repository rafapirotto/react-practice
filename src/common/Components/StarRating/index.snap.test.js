import React from 'react';
import renderer from 'react-test-renderer';

import StarRating from '.';

it('matches the snapshot', () => {
  const component = renderer.create(<StarRating filledStars={4} halfFilledStars={1} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
