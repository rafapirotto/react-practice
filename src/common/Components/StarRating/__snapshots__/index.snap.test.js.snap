// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It matches the snapshot 1`] = `
<div
  className="ml-12 flex items-center"
>
  <img
    alt="filled-star"
    className="px-1"
    src="filled-star.png"
  />
  <img
    alt="filled-star"
    className="px-1"
    src="filled-star.png"
  />
  <img
    alt="filled-star"
    className="px-1"
    src="filled-star.png"
  />
  <img
    alt="filled-star"
    className="px-1"
    src="filled-star.png"
  />
  <img
    alt="half-filled-star"
    className="px-1"
    src="semi-filled-star.png"
  />
</div>
`;
